scripts/employ-gha-switches.R

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env Rscript
+
+library(magrittr)
+
+# Make extract trigger function
+extract_trigger <- function(gha_contents) {
+  # Extract trigger criteria
+  trigger_start <- grep("TRIGGER-START", gha_contents)
+  trigger_end <- grep("TRIGGER-END", gha_contents)
+
+  return(trigger_start:trigger_end)
+}
+
+# Find .git root directory
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".github"))
+
+### Load in "on" triggers
+
+# Each github action has its own
+on_triggers <- readr::read_rds(file.path(root_dir, ".github", "on-triggers.rds"))
+
+# Same off trigger for all
+off_trigger <- c("#---TRIGGER-START---#",
+                 "on:",
+                 "  workflow_dispatch:",
+                 "#---TRIGGER-END----#")
+
+### Read in and set up config automation yaml
+
+# Get github actions directory
+github_actions_dir <- file.path(root_dir, ".github", "workflows")
+
+# Get list of github actions files
+github_actions_files <- list.files(github_actions_dir, pattern = "\\.yml$",
+                                   full.names = TRUE)
+
+# Set up the list of the gha
+config_yaml <- yaml::read_yaml(file.path(root_dir, "config_automation.yml")) %>%
+  data.frame() %>%
+  t() %>%
+  data.frame() %>%
+  tibble::rownames_to_column("gha_files") %>%
+  dplyr::mutate(gha_files = paste0(gsub("\\.", "-", gha_files), ".yml"))
+
+colnames(config_yaml)[2] <- "on_or_off"
+
+# Read in all files
+all_gha <- lapply(github_actions_files, function(gha_file) {
+
+  # Read in yaml
+  yaml_contents <- readLines(gha_file)
+
+  # Should this be on or off?
+  status <- config_yaml %>%
+    dplyr::filter(basename(gha_file) == gha_files) %>%
+    dplyr::pull("on_or_off")
+
+  if (length(status) == 1) {
```

Comment:
It's also bad that the trigger gets overwritten whether or not there is a change specified in `config_automation.yml`

Reply:
I think my next idea for this, and for the other problem is what if we store a yaml of all the triggers. (`.github/triggers.yml`) 
So for example: 
```
check-quizzes: 
  status: on 
  on: <What the trigger for on looks like>

  off: <What the trigger for off looks like>

render-bookdown: 
  status: on
  on: < ... > 

  off: < ... > 
```

This format would be far easier to manage than an RDS because we can 1) Just change the triggers in this file directly using a text editor and 2) We can use it to check the current status with the previous status so we can avoid overwriting everytime.

Reply:
My hope is that this kind of yaml would allow the `store-triggers.R` script render unnecessary completely and would make everything a lot more streamlined.

Reply:
with the latest iteration of this, `store-triggers.R` is not obsolete and now you just need to specify things in the `gha-triggers.yml` if you want to change the triggers.